osutil/user.go

```diff
@@ -210,14 +217,24 @@ func AddUser(name string, opts *AddUserOptions) error {
 		"--gecos", opts.Gecos,
 		"--disabled-password",
 	}
+	if !hasAddUserExecutable() {
+		cmdStr = []string{
+			"useradd",
+			"--badname",
```

Comment:
--badname and --force-badname do not have quite the same affect in useradd and adduser, we need to check carefully what we want to allow. Afaict  useradd allows something like "A~b"  but adduser doesn't even with --force-badname

Reply:
As discussed, updated in 814ae89fca1fee024d7dbff3de623b6701033362